### PR TITLE
Fix/version table

### PR DESCRIPTION
**Pull request by bartj3**
This PR fixes the missing table after the CI would deploy a new version. It also adds improvements to the flow to make it faster and updates the dependencies to their latest version.

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -10,8 +9,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
-      - name: Build & Deploy to GitHub Pages
+      - uses: ruby/setup-ruby@v1
         with:
-          REMOTE_BRANCH: gh-pages
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        uses: yurikoval/middleman-gh-pages-action@master
+          ruby-version: 3.1.2
```
Comment (joeyhorst) [inline .github/workflows/gh-pages.yml:14]: Doesn't this break anything else since it was always mentioned to use 2.6.9?

**Review by joeyhorst — APPROVED** [review]

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (bartj3) [reply .github/workflows/gh-pages.yml:14]: At some point we need to upgrade, we can't stay on old versions forever. The newer Ruby version is what caused the issue with the API table but that's now resolved.

3.1 isn't the latest Ruby version but recent enough that it'll be supported for a while (march 2025).

**Comment by joeyhorst** [comment]
I'll pull this branch to see if it solves https://github.com/GRESB/api-docs/pull/115 as well 👍

**Comment by joeyhorst** [comment]
It keeps hanging on `sass-rails (6.0.0)` for me - waited 30min for it to build.

```
joeyhorst@joey-horst-msi-creator-z16:~/api-docs$ docker compose build
[+] Building 3091.6s (11/12)
 => [internal] load build definition from Dockerfile                                                            0.0s
 => => transferring dockerfile: 32B                                                                             0.0s
 => [internal] load .dockerignore                                                                               0.0s
 => => transferring context: 2B                                                                                 0.0s
 => [internal] load metadata for docker.io/library/ruby:3.1.2-alpine3.14                                        3.0s
 => [auth] library/ruby:pull token for registry-1.docker.io                                                     0.0s
 => [internal] load build context                                                                               0.0s
 => => transferring context: 61B                                                                                0.0s
 => [1/7] FROM docker.io/library/ruby:3.1.2-alpine3.14@sha256:876df894fdb2c3c8e829e8dff2e12838363332716563af9f  0.0s
 => CACHED [2/7] WORKDIR /app                                                                                   0.0s
 => CACHED [3/7] COPY Gemfile* ./                                                                               0.0s
 => CACHED [4/7] RUN gem install bundler:2.4.1                                                                  0.0s
 => CACHED [5/7] RUN apk add --update --no-cache                      build-base                      python3   0.0s
 => CACHED [6/7] RUN bundle config force_ruby_platform true                                                     0.0s
 => [7/7] RUN bundle install --jobs 4 --retry 3                                                              3001.6s
 => => # Installing sprockets-rails 3.4.2
 => => # Installing railties 7.0.4
 => => # Fetching sassc-rails 2.1.2
 => => # Installing sassc-rails 2.1.2
 => => # Fetching sass-rails 6.0.0
 => => # Installing sass-rails 6.0.0
 ```

**Pull request merged**